src/jailer/src/cgroup.rs

```diff
@@ -125,114 +116,129 @@ fn inherit_from_parent(path: &mut PathBuf, file_name: &str) -> Result<()> {
     inherit_from_parent_aux(path, file_name, true)
 }
 
+// This function should be removed once the `--node` argument is removed.
+// This function generates the corresponding cgroups for isolating the process in the specified
+// NUMA node.
+pub fn cgroups_from_numa_node(
+    numa_node: u32,
+    microvm_id: &str,
+    exec_file_name: &OsStr,
+) -> Result<Vec<Cgroup>> {
+    // retrieve the CPUs which belongs to the specific node
+    // Similar to how numactl library does, we are copying the contents of
+    // /sys/devices/system/node/nodeX/cpulist to the cpuset.cpus file for ensuring
+    // correct numa cpu assignment.
+    let cpus = readln_special(&PathBuf::from(format!(
+        "{}{}/cpulist",
+        NODE_TO_CPULIST, numa_node
+    )))?;
+
+    // isolates the process in the specified numa_node CPUs.
+    let cpuset_cpus = Cgroup::new("cpuset.cpus".to_string(), cpus, microvm_id, exec_file_name)?;
+
+    // isolates the process in the specified numa_node memory.
+    let cpuset_mems = Cgroup::new(
+        "cpuset.mems".to_string(),
+        numa_node.to_string(),
+        microvm_id,
+        exec_file_name,
+    )?;
+
+    Ok(vec![cpuset_cpus, cpuset_mems])
+}
+
 impl Cgroup {
-    pub fn new(id: &str, numa_node: u32, exec_file_name: &OsStr) -> Result<Self> {
+    pub fn new(file: String, value: String, id: &str, exec_file_name: &OsStr) -> Result<Self> {
+        let cgroup_location = Self::get_location(&file, exec_file_name, id)?;
+
+        Ok(Cgroup {
+            file,
+            value,
+            location: cgroup_location,
+        })
+    }
+
+    // This writes the cgroup value into the cgroup property file.
+    pub fn write_value(&self) -> Result<()> {
+        let location = &mut self.location.clone();
+
+        // Creates the cgroup directory for the controller
+        fs::create_dir_all(&self.location)
+            .map_err(|e| Error::CreateDir(self.location.clone(), e))?;
+
+        // Writes the corresponding cgroup value. inherit_from_parent is used to
+        // correctly propagate the value if not defined
+        inherit_from_parent(location, &self.file)?;
+        location.push(&self.file);
+        writeln_special(location, &self.value)?;
+
+        // Ensure the value has been correctly written
+        let read_val = readln_special(location)?.trim().to_string();
+        if self.value != read_val {
+            return Err(Error::CgroupWrite(
+                self.value.to_string(),
+                read_val,
+                self.file.to_string(),
+            ));
+        }
+
+        Ok(())
+    }
+
+    // This writes the pid of the current process to the tasks file. Tasks files are special files,
+    // that when written to, will assign the process associated with the pid to the respective cgroup.
+    pub fn attach_pid(&self) -> Result<()> {
```

Comment:
I check that the PID is correctly written in the integration tests here: https://github.com/firecracker-microvm/firecracker/pull/2012/files?file-filters%5B%5D=#diff-de53811cc3ef46aefff70aaed734b479R130. I can try to improve it and check if the pid match, instead of just checking if it's a digit.